test-main.js

```diff
@@ -0,0 +1,65 @@
+var allTestFiles = ['assert', 'underscore'];
+var TEST_REGEXP = /tests\/.*\.js$/i;
+
+// Automatically clenaup sinon spies and stubs.
+var realSinon = sinon;
+beforeEach(function() {
+    sinon = realSinon.sandbox.create();
+});
+afterEach(function() {
+    sinon.restore();
+    sinon = realSinon;
+});
+
+function withSettings(changes, test) {
```

Comment:
Squirejs looks like it would work for this kind of case: https://github.com/iammerrick/Squire.js/